portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,27 @@ function addRandomFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
 }
+
+/**
+ * Shuffle an array in place.
+ */
+function shuffle(arr) {
+  arr.sort((a, b) => Math.random() - 0.5);
+}
+
+/**
+ * Shuffles images in the gallery.
+ */
+function shuffleImagesInGallery() {
+  const gallery = document.getElementById('gallery');
+  const images = gallery.getElementsByTagName('img');
+  
+  var indices = []
+  for(i=0; i < images.length; i++) {
+    indices.push(i);
+  }
+  shuffle(indices);
+  for(i = 0; i < images.length; i++) {
+    gallery.appendChild(images[indices[i]]);
+  }
```

Comment:
I'd rather just shuffle the images collection directly instead of shuffling the indices. Makes it shorter and easier to understand IMO. Something like this -
```suggestion
  shuffle(Array.from(images));
  for (var image of images) {
    gallery.appendChild(image);
  }
```

Reply:
Thanks for this suggestion! Regarding this append child function, actually I'm a bit confused about its behavior. It works as it moves the original element to a new place, why doesn't it make a copy?